### PR TITLE
Capture fakes.version instead of the current version string in baselines

### PR DESCRIPTION
**Pull request by weswigham**
#36474 broke our nightly publish because it added a test which captured the current version into the baselines. Naturally, this changes when a release is made, making the baseline fail.

### CODE REVIEW

File: src/testRunner/unittests/tsc/helpers.ts [hunk 1]
```diff
@@ -118,7 +118,7 @@ namespace ts {
         sys.baseLine = () => {
             const patch = fs.diff(baseFs, { includeChangedFileWithSameContent: true });
             // eslint-disable-next-line no-null/no-null
-            Harness.Baseline.runBaseline(`${isBuild(commandLineArgs) ? "tsbuild" : "tsc"}/${scenario}/${buildKind || BuildKind.Initial}/${subScenario.split(" ").join("-")}.js`, patch ? vfs.formatPatch(patch) : null);
+            Harness.Baseline.runBaseline(`${isBuild(commandLineArgs) ? "tsbuild" : "tsc"}/${scenario}/${buildKind || BuildKind.Initial}/${subScenario.split(" ").join("-")}.js`, patch ? vfs.formatPatch(patch).replace(version, "CURRENT.VERSION") : null);
```
Comment (sheetalkamat) [inline src/testRunner/unittests/tsc/helpers.ts]: Can you please change this to use fakes.version instead.. We do that in the buildinfo already. Thank you for catching this.

Comment (weswigham) [reply src/testRunner/unittests/tsc/helpers.ts]: Sure.

Comment (sheetalkamat) [reply src/testRunner/unittests/tsc/helpers.ts]: Also better yet in the `serializeOutput in virtualFileSystemWithWatch`

Comment (weswigham) [reply src/testRunner/unittests/tsc/helpers.ts]: `tscCompile` uses `fakes.System`, and not `TestServerHost`; so modifying `serializeOutput` in `virtualFileSystemWithWatch` does nothing.

Comment (weswigham) [reply src/testRunner/unittests/tsc/helpers.ts]: I've modified `fakes.patchHostForBuildInfoReadWrite` to also override `sys.write` to replace the current version with `fakes.version`. I think that should move it closer to a more general position where the versions are handled in bundle infos like what you were thinking?

**Pull request merged**